brats/losses.py

```diff
@@ -0,0 +1,38 @@
+import torch
+import torch.nn as nn
+
+
+class DiceLossOneClass(nn.Module):
+    """
+    Compute Dice Loss for single class.
+    The loss will be averaged across channels.
+    """
+
+    def __init__(self, epsilon: float = 1e-5):
+        """
+        Args:
+            epsilon: smooth factor to prevent division by zero
+        """
+        super(DiceLossOneClass, self).__init__()
+        self.epsilon = epsilon
+
+    def __call__(self, prediction: torch.Tensor,
+                 target: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            prediction: output of a network, expected to be already binarized.
+                Dimensions - (Batch, Channels, Height, Width)
+            target: labels.
+                Dimensions - (Batch, Channels, Height, Width)
+
+        Returns:
+            torch.Tensor: Computed Dice Loss
+        """
+        all_but_batch_dims = list(range(1, target.dim()))
+        prediction = prediction.contiguous()
```

Comment:
Are we sure the tensors that get into the loss function aren't contiguous? Even when they aren't do we want to make them so? I think that this requires allocating new memory and copying all the values under consideration and that might be actually more costly that working on non-contiguous tensors.

Reply:
Yeah, you could bring this back when you will be training the network and see if there is a execution time difference between case with and without. Maybe it would provide enough info to decide on the matter.

Reply:
I've added new issue for this, otherwise we would forget about it #23